tests/test_output_transformations.py

```diff
@@ -450,6 +454,42 @@ def test_get_afwijking_data_raises_type_error(
                 validation_output="123",
             )
 
+
+def test_get_highest_severity_from_validation_result():
+    validation_result = {
+        "results": [
+            {
+                "success": False,
+                "expectation_config": {"type": "expect_column_values_to_not_be_null"}
+            },
+            {
+                "success": False,
+                "expectation_config": {"type": "expect_column_values_to_be_unique"}
+            },
+            {
+                "success": True,
+                "expectation_config": {"type": "expect_column_values_to_not_be_null"}
+            }
+        ]
+    }
+ 
+    rules_dict = {
+        "rules": [
+            {
+                "rule_name": "ExpectColumnValuesToNotBeNull",
+                "parameters": {"column": "name"},
+                "severity": "warning"
+            },
+            {
+                "rule_name": "ExpectColumnValuesToBeUnique",
+                "parameters": {"column": "id"},
+                "severity": "fatal"
+            }
+        ]
+    }
+    result = get_highest_severity_from_validation_result(validation_result, rules_dict)
+    assert result == "fatal"
```

Comment:
I would also add a test to check that "ok" is returned when severity is absent in the `rules_dict` or when the results only contains `"success": True` values.

Reply:
Also, try adding a test case to `test_validation_input` catching the exception where `severity` is not equal to `None`, "fatal", "error" or "warning".